### ISSUE TITLE
please improve usage explanation

### ISSUE THREAD

**Issue by ajoslin103** [arikw/flat-promise#3]
I am unclear on how to use flatPromise in real life -- with code that does something...

It would be great to have a slightly more real-world example -- with an executor that has a function.  I have never written a Promise that didn't do something, and your flatPromise example doesn't do anything, or is that entirely the point?

perhaps: make a flat promise example that uses a setTimeout to reject internally in 2 seconds, but resolve it externally in one second?

thanks!

**Comment by arikw** [comment]
Yes, the Promise isn't doing anything in its executor function except exposing the resolution methods. The goal is to create a promise and resolve it outside of its executor function, as oppose to within the executor function, and by that maintaining a "flat" code in terms of indentation. 

Real world examples are available on various repositories on github: https://github.com/search?q=require%28%27flat-promise%27%29&type=Code
